src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Search/Expressions/Parsers/ExpressionParserTests.cs

```diff
@@ -27,14 +29,16 @@ public class ExpressionParserTests
     {
         private readonly ISearchParameterDefinitionManager _searchParameterDefinitionManager = Substitute.For<ISearchParameterDefinitionManager>();
         private readonly ISearchParameterExpressionParser _searchParameterExpressionParser = Substitute.For<ISearchParameterExpressionParser>();
+        private RequestContextAccessor<IFhirRequestContext> _contextAccessor = Substitute.For<RequestContextAccessor<IFhirRequestContext>>();
 
         private readonly ExpressionParser _expressionParser;
 
         public ExpressionParserTests()
         {
             _expressionParser = new ExpressionParser(
                 () => _searchParameterDefinitionManager,
-                _searchParameterExpressionParser);
+                _searchParameterExpressionParser,
+                _contextAccessor);
```

Comment:
Add tests to this file for the changes made in `ExpressionsParser.cs`, for the more complex cases - for example, a case where one of the targeted resource types are restricted by clinical scopes and the generated expression looks as expected (I'm guessing that it shouldn't entirely fail but only block out only the disallowed resource type?)

Reply:
Nvm, I see that `SmartSearchTests` tests this functionality. Resolving this comment